src/System.IO.IsolatedStorage/tests/System/IO/IsolatedStorage/GetCreationTimeTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.IsolatedStorage
+{
+    public class GetCreationTimeTests : IsoStorageTest
+    {
+        [Fact]
+        public void GetCreationTime_ThrowsArugmentNull()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                Assert.Throws<ArgumentNullException>(() => isf.GetCreationTime(null));
+            }
+        }
+
+        [Fact]
+        public void GetCreationTime_ThrowsIsolatedStorageException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                isf.Remove();
+                Assert.Throws<IsolatedStorageException>(() => isf.GetCreationTime("foo"));
+            }
+        }
+
+        [Fact]
+        public void GetCreationTime_ThrowsObjectDisposed()
+        {
+            IsolatedStorageFile isf;
+            using (isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+            }
+
+            Assert.Throws<ObjectDisposedException>(() => isf.GetCreationTime("foo"));
+        }
+
+        [Fact]
+        public void GetCreationTime_ThrowsInvalidOperationException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                isf.Close();
+                Assert.Throws<InvalidOperationException>(() => isf.GetCreationTime("foo"));
+            }
+        }
+
+        [Fact]
+        public void GetCreationTime_RaisesArgumentException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                Assert.Throws<ArgumentException>(() => isf.GetCreationTime("\0bad"));
+            }
+        }
+
+        [Fact]
+        public void GetCreationTime_GetsTime()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                string file = "GetCreationTime_GetsTime";
+                DateTimeOffset currentTime = DateTimeOffset.Now;
+                isf.CreateTestFile(file);
+                DateTimeOffset fileTime = isf.GetCreationTime(file);
+                TimeSpan difference = fileTime - currentTime;
+                Assert.True(difference.TotalSeconds < 1.0);
```

Comment:
Because the file times vary in granularity, caching, etc. https://msdn.microsoft.com/en-us/library/windows/desktop/ms724290(v=vs.85).aspx

The system clock is way more precise. I don't want to have to allow for all the various roundings involved across all of the file systems involved. I may have to tweak this further as it is- FAT access time is rounded to the day for example.